app/webhook.go

```diff
@@ -340,19 +340,19 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
 		return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_username.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	if len(hook.WhiteIpList) > 0 {
-		if !(model.IsValidIpList(hook.WhiteIpList)) {
-			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_white_ip_list.app_error", nil, "", http.StatusBadRequest)
+	if len(hook.IpWhiteList) > 0 {
+		if !(model.IsValidIpList(hook.IpWhiteList)) {
+			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_ip_whitelist.app_error", nil, "", http.StatusBadRequest)
 		}
+		hook.ResetSignatureModels()
 	} else if len(hook.HmacAlgorithm) > 0 {
-		if hook.IsValidModels() {
-			hook.SecretToken = model.NewRandomString(26)
-			// hook.SecretToken = "me" // for testing
-		} else {
-			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_signing_model.app_error", nil, "", http.StatusBadRequest)
+		if err := hook.IsValidSignatureModels(); err != nil {
+			return nil, err
 		}
+		hook.SecretToken = model.NewRandomString(26)
+		// hook.SecretToken = "me" // for testing
```

Comment:
Remove now that we're pushing towards final review?

Reply:
done.